File: LightGlide/app/src/main/java/cy/ac/ucy/cs/epl341/team5/lightglide/db/model/Flight.java
package cy.ac.ucy.cs.epl341.team5.lightglide.db.model;

import java.sql.Timestamp;

public final class Flight {
    final int id;
    final String name;
    final Timestamp start;
    final Timestamp end;
    final int max_altitude;
    final double distnace;
    final double max_velocity;
    final int duration;


    Flight(int id, String name, Timestamp start, Timestamp end, int max_altitude, double distnace, double max_velocity, int duration) {
        this.id = id;
        this.name = name;
        this.start = start;
        this.end = end;
        this.max_altitude = max_altitude;
        this.distnace = distnace;
        this.max_velocity = max_velocity;
        this.duration = duration;
    }
}

File: LightGlide/app/src/main/java/cy/ac/ucy/cs/epl341/team5/lightglide/db/model/Flight.kt
package cy.ac.ucy.cs.epl341.team5.lightglide.db.model

import android.os.Parcel
import android.os.Parcelable
import java.sql.Time
import java.sql.Timestamp
import java.time.Duration

data class Flight constructor(val name:String?, val start:Timestamp, val maxAltitude: Int, val distance: Double, val end: Timestamp, val duration: Int):Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.readString(),
            Timestamp(parcel.readLong()),
            parcel.readInt(),
            parcel.readDouble(),
            Timestamp(parcel.readLong()),
                    parcel.readInt())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(name)
        parcel.writeInt(maxAltitude)
        parcel.writeDouble(distance)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Flight> {
        override fun createFromParcel(parcel: Parcel): Flight {
            return Flight(parcel)
        }

        override fun newArray(size: Int): Array<Flight?> {
            return arrayOfNulls(size)
        }
    }
}